### PR TITLE
Fix Tuya signed 16 bit temperature values

### PR DESCRIPTION
**Pull request by javicalle**
Another approach to fix the 16 bits reports from some devices.
Supersed #2180
Fixes: #1976
(And will not break any sensor while temp < 4095 ºC)

### CODE REVIEW

**Comment by javicalle** [comment]
@Shulyaka I was also figuring out how can be done the same that yours PR.
Because I don't feel confortable with all the bitwise operations I searched for another approach.
I don't like all the `serialize` / `deserialize` part but I haven't found a more elegant a way to implement that.

I don't think there is one version better than another, it's more a matter of preference.

You can take the tests for your PR.

**Comment by Shulyaka** [comment]
@javicalle I like your approach more

**Comment by Shulyaka** [comment]
@TheJulianJES No, https://github.com/zigpy/zha-device-handlers/pull/2176 is not related.

**Comment by javicalle** [comment]
Any idea how can be done to parse the value directly?

What I would have is:
* a 'CustomTemperature' definition like the one defined:
```python
class CustomTemperature(t.Struct):
    """Custom temperature wrapper."""

    field_1: t.int16s_be
    temperature: t.int16s_be
```
* define the attribute type with that class:
```python
            0xEF12: ("custom_temperature", CustomTemperature, False),
```
* handle the value directly without conversions:
```python
    def update_attribute(self, attr_name: str, value: Any) -> None:
        """Calculate the current temperature."""

        self._update_attribute(self.attributes_by_name[attr_name].id, value)

        if attr_name == "custom_temperature":
            custom_temperature = self.attributes_by_name["custom_temperature"]  # that would be an CustomTemperature instance
            if custom_temperature:
                temp = custom_temperature.temperature * 10
                super().update_attribute("measured_value", temp)
```

I have tried all the data types that I can but I didn't get any clean solution.

**Comment by javicalle** [comment]
Ready to review now.
Not sure if better to maintain (and extend). 
Thanks @Shulyaka!

File: zhaquirks/tuya/air/__init__.py [hunk 1]
```diff
@@ -31,9 +31,42 @@ class TuyaAirQualityVOC(TuyaLocalCluster):
...
+
+    def update_attribute(self, attr_name: str, value: Any) -> None:
+        """Calculate the current temperature."""
+
+        super().update_attribute(attr_name, value)
+
+        if attr_name == "custom_temperature":
+            custom_temperature = self._attr_cache.get(
+                self.attributes_by_name[attr_name].id,
+            )
```
Comment (TheJulianJES) [inline zhaquirks/tuya/air/__init__.py:65]: Not that it matters, but what's the benefit of getting `custom_temperature` from the attribute cache again when it should also be `value` here?

Comment (javicalle) [reply zhaquirks/tuya/air/__init__.py]: You are right!!! 
I was expecting that value was the raw value (so no `value.temperature`) and we have to recover from class 😅 
Just checked and working like a charm

Thanks Julian!

**Review by TheJulianJES — APPROVED** [review]
LGTM

**Pull request merged**